invenio_files_rest/storage.py

```diff
@@ -127,3 +132,83 @@ def send_file(self, restricted=False):
                 content_md5=self.file.checksum)
         except Exception as e:
             raise StorageError('Could not send file: {}'.format(e))
+
+
+class AmazonS3(Storage):
+    def __init__(self, object, fileinstance):
+        super(AmazonS3, self).__init__(object, fileinstance)
+        if "AWS_KEY" in current_app.config:
+            raise StorageError("Missing AWS Key")
+        if "AWS_SECRET" in current_app.config:
+            raise StorageError("Missing AWS Secret")
+        self.aws_connection = S3Connection(current_app.config.get("AWS_KEY", "AWS_KEY"),
+                                           current_app.config.get("AWS_SECRET", "AWS_SECRET"))
+
+    def get_size(self):
+        bucket_name, key_name = self.obj.file.uri.replace("s3:", "").split(":")
+        bucket = self.aws_connection.get_bucket(bucket_name)
+        return bucket.lookup(key_name).size
+
+    def open(self):
+
+        temp = tempfile.NamedTemporaryFile()
+        try:
+            loc = self.obj.bucket.location.uri
+            bucket = self.aws_connection.get_bucket(loc)
+            element = Key(bucket)
+            key_name = self.obj.file.uri.replace("s3:", "").split(":")[1]
+            element.key = key_name
+            element.get_contents_to_file(temp)
+            temp.seek(0)
+            return temp
+        except:
```

Comment:
Try to avoid ``except`` without an exception unless you really want to catch keyboard interrupts etc.